osu.Game/Screens/Edit/GameplayTest/EditorPlayer.cs

```diff
@@ -173,27 +173,53 @@ public void OnReleased(KeyBindingReleaseEvent<GlobalAction> e)
         {
         }
 
-        private void toggleAutoplay()
+        private double previousAutoTime = 0;
+
+        private void setupAutoplay()
         {
-            if (DrawableRuleset.ReplayScore == null)
-            {
-                var autoplay = Ruleset.Value.CreateInstance().GetAutoplayMod();
-                if (autoplay == null)
-                    return;
+            var autoplay = Ruleset.Value.CreateInstance().GetAutoplayMod();
+            if (autoplay == null)
+                return;
+
+            // stores time when autoplay starts
+            previousAutoTime = GameplayClockContainer.CurrentTime;
 
-                var score = autoplay.CreateScoreFromReplayData(GameplayState.Beatmap, [autoplay]);
+            var score = autoplay.CreateScoreFromReplayData(GameplayState.Beatmap, new[] { autoplay });
 
-                // remove past frames to prevent replay frame handler from seeking back to start in an attempt to play back the entirety of the replay.
-                score.Replay.Frames.RemoveAll(f => f.Time <= GameplayClockContainer.CurrentTime);
+            // remove past frames to prevent replay frame handler from seeking back to start in an attempt to play back the entirety of the replay.
+            score.Replay.Frames.RemoveAll(f => f.Time <= GameplayClockContainer.CurrentTime);
 
-                DrawableRuleset.SetReplayScore(score);
-                // Without this schedule, the `GlobalCursorDisplay.Update()` machinery will fade the gameplay cursor out, but we still want it to show.
-                Schedule(() => DrawableRuleset.Cursor?.Show());
+            DrawableRuleset.SetReplayScore(score);
+            // Without this schedule, the `GlobalCursorDisplay.Update()` machinery will fade the gameplay cursor out, but we still want it to show.
+            Schedule(() => DrawableRuleset.Cursor?.Show());
+        }
+
+        private void toggleAutoplay()
+        {
+            if (DrawableRuleset.ReplayScore == null)
+            {
+                setupAutoplay();
             }
             else
                 DrawableRuleset.SetReplayScore(null);
         }
 
+        protected override void Update()
+        {
+            // updates if there's a replay (autoplay)
+            if (DrawableRuleset.ReplayScore != null)
+            {
+                // compare currentTime with previousTime
+                if (GameplayClockContainer.CurrentTime != previousAutoTime)
+                {
+                    if (GameplayClockContainer.CurrentTime < previousAutoTime)
```

Comment:
Why are these checks nested in each other like that? Only the inner one suffices.

Reply:
understood